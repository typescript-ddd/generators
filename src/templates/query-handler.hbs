import { QueryHandler } from "@typescript-ddd/core";
import { {{pascal entityName}}View } from "../../domain/views";
import { {{pascal entityName}}Query } from ".";

/**
 * Represents a handler for {{noun entityName}} query.
 */
export class {{pascal entityName}}QueryHandler implements QueryHandler<{{pascal entityName}}Query> {
  /**
   * Initializes a new {{natural entityName}} query handler instance.
   * @param {{lbrace}}{{pascal entityName}}Finder{{rbrace}} {{lower entityName}}Finder - the {{natural entityName}} finder.
   */
  constructor(private readonly {{lower entityName}}Finder: {{pascal entityName}}Finder) {}

  /**
   * Handles the {{natural entityName}} query.
   * @param {{lbrace}}{{pascal entityName}}Query{{rbrace}} query - The query to handle.
   * @returns {{lbrace}}{{pascal entityName}}View{{rbrace}}
   */
  public async handle(query: {{pascal entityName}}Query): Promise<{{pascal entityName}}View> {
    return await this.{{lower entityName}}Finder.find({
      {{#each properties}}
      {{name}}: query.{{name}},
      {{/each}}
    });
  }
}
