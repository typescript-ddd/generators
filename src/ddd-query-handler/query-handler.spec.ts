import { generateQueryHandler} from "./query-handler.generator";
import { QueryHandlerTemplateData } from "./query-handler.types";

describe("CommandHandler", () => {
  it("should render a template", () => {
    // Arrange
    const data: QueryHandlerTemplateData = {
      entityName: "User",
      properties: [
        { name: "name", type: "string" },
        { name: "email", type: "string" },
      ],
    };

    // Act
    const result = generateQueryHandler(data);

    // Assert
    expect(result).toMatchSnapshot();
  });
});
