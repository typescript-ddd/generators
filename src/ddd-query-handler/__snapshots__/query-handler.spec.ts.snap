// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CommandHandler should render a template 1`] = `
"import { QueryHandler } from "@typescript-ddd/core";
import { UserView } from "../../domain/views";
import { UserQuery } from ".";

/**
 * Represents a handler for a user query.
 */
export class UserQueryHandler implements QueryHandler<UserQuery> {
  /**
   * Initializes a new user query handler instance.
   * @param {UserFinder} userFinder - the user finder.
   */
  constructor(private readonly userFinder: UserFinder) {}

  /**
   * Handles the user query.
   * @param {UserQuery} query - The query to handle.
   * @returns {UserView}
   */
  public async handle(query: UserQuery): Promise<UserView> {
    return await this.userFinder.find({
      name: query.name,
      email: query.email,
    });
  }
}
"
`;
